release-notes/opensearch-release-notes-3.1.0.md

```diff
@@ -0,0 +1,852 @@
+* Immutable user object for security authorization: The user object is now immutable in the Security plugin, reducing the number of serializations and deserializations required as this object is passed internally within the cluster while handling a request, resulting in lower performance overhead.
+* Privilege optimization for tenants: This release incorporates updates to privilege evaluation as part of an ongoing optimization effort. Extending privilege evaluation to encompass tenant_permissions lets OpenSearch roles take full advantage of precomputed data structures for performance improvements, particularly for clusters using multi-tenancy with a large number of tenants.
+* Collapse in hybrid query: This release extends hybrid query functionality with collapse parameter support, enabling document grouping and deduplication based on specified field values.
+* New encoding of BPV 21 for DocIdsWriter: A new variation of an existing encoding algorithm offers optimized docId storage within the BKD index, resulting in reduced storage footprint across numeric, IP, and other fields using the BKD index while facilitating more efficient docId retrieval.
+
+### Experimental Features
+
+OpenSearch 3.1 includes the following experimental functionality. Experimental features are disabled by default. For instructions on how to enable them, refer to the documentation for the feature.
+ 
+* Resource sharing and access control: An experimental authorization framework moves the sharing and access authorization setup from individual plugins to the Security plugin to support improved security posture. Plugins will be required to onboard to this new feature; for this release, the Anomaly Detection plugin has been updated to support the new authorization framework.
+* Model Context Protocol (MCP) enhancements: New update MCP tools and list MCP tools APIs add functionality to OpenSearch's MCP support. A new system index is also included, allowing the MCP tools to persist in the system index so that tools won't be lost after a restart at the cluster or node level.
+
+
+## Release Details
+[OpenSearch and OpenSearch Dashboards 3.1.0](https://opensearch.org/artifacts/by-version/#release-3-1-0) includes the following breaking changes, features, enhancements, bug fixes, infrastructure, documentation, maintenance and refactoring updates.
+
+OpenSearch [Release Notes](https://github.com/opensearch-project/OpenSearch/blob/main/release-notes/opensearch.release-notes-3.1.0.md).
+
+OpenSearch Dashboards [Release Notes](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/release-notes/opensearch-dashboards.release-notes-3.1.0.md).
+
+
+## BREAKING CHANGES
+
+
+### SQL
+
+
+* Switch percentile implementation to MergingDigest to align with OpenSearch ([#3698](https://github.com/opensearch-project/sql/pull/3698))
+* Support decimal literal with Calcite ([#3673](https://github.com/opensearch-project/sql/pull/3673))
```

Comment:
Can you confirm with SQL is this is actually breaking?

Since 3.1.0 is minor release.
Please feel free to tag the owner